docs/backup-restore-integration.md

```diff
@@ -0,0 +1,287 @@
+
+#### spec.template
+
+See [VirtualMachineInstance](#virtualmachineinstance-object-graph)
+
+### VirtualMachineInstance Object Graph
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: vmi1
+  namespace: ns1
+...
+```
+
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmi1")
+- ("", "Pod", "ns1", "virt-launcher-vmi1-XXXXX") \*
+
+\* Each VirtualMachineInstance has a corresponding uniquely named Pod.  The backup process can look up the name of this pod by using the `kubevirt.io/created-by=<UID of VirtualMachineInstance>` label selector.
+
+#### spec.volumes[\*].persistentVolumeClaim
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    persistentVolumeClaim:
+      claimName: pvc1
+...
+```
+
+- ("", "PersistentVolumeClaim", "ns1", "pvc1")
+- ("cdi.kubevirt.io", "DataVolume", "ns1", "pvc1")
+
+#### spec.volumes[\*].dataVolume
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    dataVolume:
+      name: dv1
+...
+```
+
+- ("", "PersistentVolumeClaim", "ns1", "dv1")
+- ("cdi.kubevirt.io", "DataVolume", "ns1", "dv1")
+
+#### spec.volumes[\*].configMap
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    configMap:
+      name: cm1
+...
+```
+
+- ("", "ConfigMap", "ns1", "cm1")
+
+#### spec.volumes[\*].secret
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    secret:
+      secretName: s1
+...
+```
+
+- ("", "Secret", "ns1", "s1")
+
+#### spec.volumes[\*].serviceAccount
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    serviceAccount:
+      serviceAccountName: sa1
+...
+```
+
+- ("", "ServiceAccount", "ns1", "sa1")
+
+#### spec.volumes[\*].memoryDump
+
+```yaml
+...
+spec:
+  volumes:
+  - name: v1
+    memoryDump:
+      claimName: pvc1
+...
+```
+
+- ("", "PersistentVolumeClaim", "ns1", "pvc1")
+
+#### spec.accessCredentials[\*].sshPublicKey
+
+```yaml
+...
+spec:
+  accessCredentials:
+  - sshPublicKey:
+      source:
+        secret:
+          secretName: my-pub-key
+...
+```
+
+- ("", "Secret", "ns1", "my-pub-key")
+
+#### spec.accessCredentials[\*].userPassword
+
+```yaml
+...
+spec:
+  accessCredentials:
+  - userPassword:
+      source:
+        secret:
+          secretName: my-user-password
+...
+```
+
+- ("", "Secret", "ns1", "my-user-password")
+
+### VirtualMachineInstanceReplicaSet Object Graph
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstanceReplicaSet
+metadata:
+  name: vmirs1
+  namespace: ns1
+...
+```
+
+- ("kubevirt.io", "VirtualMachineInstanceReplicaSet", "ns1", "vmirs1")
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmirs1XXXX1") \*
+- ("kubevirt.io", "VirtualMachineInstance", "ns1", "vmirs1XXXX2") \*
+
+\* there are usually multiple VirtualMachineInstances corresponding to a VirtualMachineInstanceReplicaSet.  The backup process can look up the name of this pod by using the `kubevirt.io/vmReplicaSet=<name of VirtualMachineInstanceReplicaSet>` label selector.
+
+#### spec.template
+
+See [VirtualMachineInstance](#virtualmachineinstance-object-graph)
```

Comment:
This and `#### spec.template` is already mentioned in lines 86-88

Reply:
241 is `VirtualMachineInstanceReplicaSet` the previous one is for `VirtualMachine` both have a `spec.template`